archive/display/src/main/scala/uk/ac/wellcome/platform/archive/display/DisplayIngestType.scala

```diff
@@ -0,0 +1,36 @@
+package uk.ac.wellcome.platform.archive.display
+
+import io.circe.CursorOp.DownField
+import io.circe.{Decoder, DecodingFailure, Encoder, Json}
+
+sealed trait DisplayIngestType { val id: String }
+
+object CreateDisplayIngestType extends DisplayIngestType {
+  override val id: String = "create"
+}
+
+object DisplayIngestType {
+
+  implicit val decoder
+    : Decoder[DisplayIngestType] = Decoder.instance[DisplayIngestType](cursor =>
+    for {
+      id <- cursor.downField("id").as[String]
+      ingestType <- id match {
+        case CreateDisplayIngestType.id => Right(CreateDisplayIngestType)
+        case _ =>
+          val fields = DownField("id") +: cursor.history
+          Left(DecodingFailure(
+            s"invalid value supplied, valid values are: ${CreateDisplayIngestType.id}.",
```

Comment:
Rather than `value`, maybe this could be `id`? And include whatever the ID the user passed was. For example:

> Invalid ID 'foobar'; expected one of "create", "destroy", "preserve".

Because this message could get confusing if you pass multiple keys/values, e.g.

```json
{
    "id": "foo",
    "name": "bar"
}
```

Which of these is the invalid value?